kb(ContextMenu): add kb for disable focus on first item (#2054)
telerik/blazor-docs: knowledge-base/contextmenu-disable-focus.md

---
title: Disable Focus on First Item in ContextMenu
description: Learn how to prevent the first item in a context menu from automatically gaining focus by disabling or removing its default focus behavior.
type: how-to
page_title: Disable Focus on First Item in ContextMenu
slug: contextmenu-kb-disable-focus
tags: context-menu
ticketid: 1622761, 1635324
res_type: kb
---

## Environment

<table>
    <tbody>
        <tr>
            <td>Product</td>
            <td>ContextMenu for Blazor</td>
        </tr>
    </tbody>
</table>

## Description

> * How to remove autofocus from the first menu item?

Reviewer: Introduce the questions.
```suggestion
This Knowledge Base article answers the following questions:
* How to remove autofocus from the first menu item?
```

* How to prevent focus on the first item of the menu?

> * How to disable the focus on first context menu item?

Reviewer: ```suggestion
* How to disable the focus on the first ContextMenu item?
```

> * How to stop ContextMenu recieves focus automaticallly when it shows?

Reviewer: ```suggestion
* How to prevent the ContextMenu from automatically receiving focus when it shows?
```

## Solution

1. Set a custom CSS class to the ContextMenu through the `Class` parameter. This configuration will allow you to target specific ContextMenu instances.

> 2. Use the defiend class to [Override the theme styles](https://docs.telerik.com/blazor-ui/styling-and-themes/override-theme-styles) by setting the CSS property **box-shadow** to **none**.

Reviewer: Elements of the programming code and markup elements are rendered using the `monospace` font and not in **bold**. See the[ formatting guidelines](https://docs.telerik.com/style-guide/element-formatting#guidelines).
```suggestion
2. Use the defiend class to [Override the theme styles](https://docs.telerik.com/blazor-ui/styling-and-themes/override-theme-styles) by setting the CSS property `box-shadow` to `none`.
```

> > The ContextMenu receives focus automatically when it shows. This behavior is a [requirement for accessible user experience](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role). If the ContextMenu doesn't receive focus automatically, keyboard users will be unable to navigate the items. Removing that focus is **not recommended becasue its violate the accessibility compliance of your application**. 

Reviewer: * Use bold sparingly and avoid using it for emphasis. See https://docs.telerik.com/style-guide/element-formatting#guidelines.
* To capture the attention of the reader, move the recommendation to the beginning of the note.
```suggestion
> Removing the ContextMenu focus is not recommended because it violates the accessibility compliance of your application. The ContextMenu receives focus automatically when it shows. This behavior is a [requirement for accessible user experience](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role). If the ContextMenu doesn't receive focus automatically, keyboard users will be unable to navigate the items. 
```

description: Learn how to prevent the first item in a context menu from automatically gaining focus by disabling or removing its default focus behavior.
type: how-to
page_title: Disable Focus on First Item in ContextMenu
slug: contextmenu-kb-disable-focus
tags: context-menu
ticketid: 1622761, 1635324
res_type: kb
---

## Environment

<table>
    <tbody>
        <tr>
            <td>Product</td>
            <td>ContextMenu for Blazor</td>
        </tr>
    </tbody>
</table>

## Description
This Knowledge Base article answers the following questions:

* How to remove autofocus from the first menu item?
* How to prevent focus on the first item of the menu?
* How to disable the focus on the first ContextMenu item?
* How to prevent the ContextMenu from automaticallly receiving focus when it shows?

## Solution

1. Set a custom CSS class to the ContextMenu through the `Class` parameter. This configuration will allow you to target specific ContextMenu instances.
2. Use the defiend class to [Override the theme styles](https://docs.telerik.com/blazor-ui/styling-and-themes/override-theme-styles) by setting the CSS property `box-shadow` to `none`.

> > Removing that focus is not recommended becasue its violate the accessibility compliance of your application. The ContextMenu receives focus automatically when it shows. This behavior is a [requirement for accessible user experience](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role). If the ContextMenu doesn't receive focus automatically, keyboard users will be unable to navigate the items. 

Reviewer: ```suggestion
> Removing the ContextMenu focus is not recommended because it violates the accessibility compliance of your application. The ContextMenu receives focus automatically when it shows. This behavior is a [requirement for accessible user experience](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role). If the ContextMenu doesn't receive focus automatically, keyboard users will be unable to navigate the items. 
```
